src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,91 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final int NAME_COL_INDEX = 0;
+    private static final int AMOUNT_COL_INDEX = 1;
+    private static final String SUPPLY_ACTION = "supply";
+    private static final String BUY_ACTION = "buy";
+    private static final String RESULT = "result";
+    private static final String[] OUTPUT_FIELDS = {SUPPLY_ACTION, BUY_ACTION, RESULT};
+    private static final String SEPARATOR = ",";
+    private static final String LINE_SEPARATOR = System.lineSeparator();
+
     public void getStatistic(String fromFileName, String toFileName) {
+        String[] dataEntries = getContentFromFile(fromFileName);
+        int[] totals = new int[OUTPUT_FIELDS.length];
+
+        for (String entry : dataEntries) {
+            String[] entryData = entry.split(SEPARATOR);
+            String action = entryData[NAME_COL_INDEX];
+            int amount = Integer.parseInt(entryData[AMOUNT_COL_INDEX]);
+
+            int actionIndex = getActionIndex(action);
+            int resultIndex = getActionIndex(RESULT);
+            totals[actionIndex] += amount;
+            totals[resultIndex] = action.equals(SUPPLY_ACTION) ? totals[resultIndex] + amount
+                    : totals[resultIndex] - amount;
+        }
+
+        writeContentToFile(totals, toFileName);
+    }
+
+    private int getActionIndex(String actionName) {
```

Comment:
Change the ordering of your methods according to it's first calls. It should be easy to read the code like a book from up to down.

Reorder so:
- getContentFomrFile
- getActionIndex